### DISCUSSION TITLE
Any future support for React Native (expo) ?

### DISCUSSION THREAD

**Discussion by octalpixel** [chakra-ui/zag#116] [General]
# 🚀 Feature request

Would there be support for expo/react-native in the future?  Wondering if it's possible given the architecture

**Comment by segunadebayo** [comment]
Hi @octalpixel,

At the moment, we don't plan to support React Native but I imagine that we can make this work easily once we get the modeling of the component correctly.

I'll move this to a discussion in the meantime.

**Comment by dohomi** [comment]
This would be a great addition.

**Comment by JCcastagne** [comment]
Hoping to see this eventually roll out!

**Comment by josephfh** [comment]
Zag looks perfect for a cross platform UI. The main challenge is that React Native does not have a defined document as its rootNode, so I guess we'd have to refactor how the rootNode is found when scope is created (createScope()) and make the typings more agnostic. 

Once that was done I imagine it would be easy to add a react native framework based on the existing react one.

I'm not familiar enough with how Zag initialises its scope to get a proof of concept working, but I'd like to help in the future if someone else can break the ice :)

**Comment by jrobber** [comment]
Yes please! Libraries like react-native-web and react-native-reusables may provide some inspiration on how to.
I was excited to try this, but we're ReactNative first so I couldn't even see if the library is good because I don't have any web only projects.

**Comment by dan-scrumsdotcom** [comment]
Expo 52 supports React DOM components. Hint hint, nudge nudge...

**Comment by dustinkatterbach** [comment]
i would Love this!

**Comment by segunadebayo** [comment]
Closing this. We don't have plans to support React Native.
